src/kixi/heimdall/components/database.clj

```diff
@@ -82,6 +85,12 @@
       (if (coll? result)
         (map (partial into {}) reformatted)
         reformatted)))
+  (update! [this table what where]
+    (exec this (hayt/update table (hayt/set-columns
+                                   (reduce-kv #(assoc %1 (util/hyphen->underscore %2) %3)
```

Comment:
Not sure you need `reduce-kv` here - the [replacer](https://github.com/MastodonC/kixi.heimdall/blob/dcc6b479a1958dcf4222cc30de8c0376f6363aaa/src/kixi/heimdall/util.clj) should cope with maps.